tracing/tck/src/main/java/org/eclipse/microprofile/telemetry/tracing/tck/BasicHttpClient.java

```diff
@@ -65,13 +67,47 @@ public int get(String path) {
             try {
                 return connection.getResponseCode();
             } catch (Exception e) {
-                throw new RuntimeException("Exception retriving " + spanUrl, e);
+                throw new RuntimeException("Exception retrieving " + spanUrl, e);
             } finally {
                 connection.disconnect();
             }
         } catch (Exception e) {
-            throw new RuntimeException("Exception retriving path " + path, e);
+            throw new RuntimeException("Exception retrieving path " + path, e);
         }
     }
 
-}
+    /**
+     * Makes a GET request to a path and returns the response code
+     *
+     * @param path
+     *            the path to request, relative to the baseUrl
+     * @return the response message
+     */
+    public String getResponseMessage(String path) {
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+        try {
+            URL spanUrl = baseUri.resolve(path).toURL();
+            HttpURLConnection connection = (HttpURLConnection) spanUrl.openConnection();
+            try {
+                BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+                String inputLine;
+                StringBuffer response = new StringBuffer();
+
+                while ((inputLine = in.readLine()) != null) {
+                    response.append(inputLine);
+                }
+                in.close();
+                return response.toString();
+            } catch (Exception e) {
+                throw new RuntimeException("Exception retrieving " + spanUrl, e);
+            } finally {
+                connection.disconnect();
+            }
```

Comment:
```suggestion
            URL spanUrl = baseUri.resolve(path).toURL();
            HttpURLConnection connection = (HttpURLConnection) spanUrl.openConnection();
            try (BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
                String inputLine;
                StringBuilder response = new StringBuilder();

                while ((inputLine = in.readLine()) != null) {
                    response.append(inputLine);
                }
                return response.toString();
            } catch (Exception e) {
                throw new RuntimeException("Exception retrieving " + spanUrl, e);
            } finally {
                connection.disconnect();
            }
```

* Open `BufferedReader` within a try-with-resources block to ensure it is always closed
* Use `StringBuilder` rather than `StringBuffer`